base/linalg/triangular.jl

```diff
@@ -324,20 +325,21 @@ function scale!{T<:Union{LowerTriangular, UnitLowerTriangular}}(A::LowerTriangul
             @inbounds A[i,j] = c * B[i,j]
         end
     end
+    return A
 end
+scale!(A::Union{UpperTriangular,LowerTriangular},c::Number) = scale!(A,A,c)
+scale!(c::Number, A::Union{UpperTriangular,LowerTriangular}) = scale!(A,c)
 
 for (t1, t2) in ([:UnitLowerTriangular, :LowerTriangular],
                  [:UnitUpperTriangular, :UpperTriangular])
-    @eval begin
-        function scale!{T,S<:Number}(A::$(t1){T}, c::S)
-            D = $(t2)(Array(promote_type(T,S), size(A)...));
-            scale!(D,A,c);
-            return D
-        end
-        function scale!{T,S<:Number}(A::$(t2){T}, c::S)
-            D = $(t2)(Array(promote_type(T,S), size(A)...));
-            scale!(D,A,c);
-            return D
+    for (type1, type2) in ([:Real, :Complex], [:Any, :Number])
+        @eval begin
+            function scale{T<:$(type1),S<:$(type2)}(A::$(t1){T}, c::S)
```

Comment:
I guess this could simply be defined as `scale(A::AbstractTriangular, x::Number)=A*x`. Actually, we should consider if the scale function without exclamation mark makes sense at all. Why not just use `*`? However, that is a discussion for another issue.